### PR TITLE
Add some basic integrationtests

### LINKED ISSUE #70: Add integration testing script
To be able to properly validate the machine-controller is working as intended, we need some kind of integration testing.

Because it is not possible to both test external PRs automatically and be sure they are not used to steal credentials, this script is not supposed to be executed automatically. Instead it will:

* Take credentials for a cloud provider, e.G. from environment
* Take a ssh pubkey
* Create a single node k8s cluster via kubeadm at cloudprovider
* Deploy machine-controller in a version built from git HEAD into the newly created cluster using the deployment in the repo
* Verify machine-controller is running
* Provide a teardown functionality

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Adds some code to allow for integration testing by creating a single-node-kubernetes cluster and deploying the `machine-controller` onto it.

**Which issue(s) this PR fixes**:
Fixes #70 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by mrIncompetent** [comment]
How can we do a cleanup?

**Comment by alvaroaleman** [comment]
`make destroy` destroys all the ressources.

I'll extend this a little to also create and verify actual node creation using @p0lyn0mial verify tool

**Comment by mrIncompetent** [comment]
https://media.giphy.com/media/l0HUcfpjmD77zuSze/giphy.gif

**Comment by p0lyn0mial** [comment]
> I'll extend this a little to also create and verify actual node creation using @p0lyn0mial verify tool

fingers crossed :)

**Comment by alvaroaleman** [comment]
So, integration finished, this now also tests the actual node creation and deletion. The latter seems somewhat flaky, often times the cloud server gets destroyed but the associated k8s node is still present.

File: test/tools/verify/cmd/main.go [hunk 1]
```diff
@@ -27,10 +30,17 @@ func main() {
 	var manifestPath string
 	var parameters string
 	var kubeConfig string
+	var createOnly bool
 
-	flag.StringVar(&kubeConfig, "kubeconfig", "", "a path to the kubeconfig.")
+	defaultKubeconfigPath, err := getDefaultKubeconfigPath()
+	if err != nil {
+		log.Fatalf("Error getting default path for kubeconfig: '%v'", err)
```
Comment (mrIncompetent) [inline test/tools/verify/cmd/main.go]: insteaf of using log, please use glog, as its the library used by the kubernetes go client

**Comment by alvaroaleman** [comment]
@mrIncompetent Already did create an issue, updated the verify tool to use glog

**Pull request merged**